### PR TITLE
Docs overhaul

### PR DESCRIPTION
**Pull request by zmc**
This PR aims to both restructure documentation, and to add a more useful introduction for testers who simply want to run existing tests.

It should also provide a better experience for people browsing HTML docs at http://docs.ceph.com/teuthology/docs/.

A current preview of HTML docs from this branch are at http://qa-proxy.ceph.com/teuthology/teuth_docs/

### CODE REVIEW

File: docs/intro_testers.rst [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+<http://ceph.github.io/sepia/adding_users/>`__.
+
+
+Terminology
+===========
+
+In the abstract, each set of tests is defined by a `suite`. All of our suites
+live in the `ceph-qa-suite repository
+<https://github.com/ceph/ceph-qa-suite/>`__, in the `suites` subdirectory.
+
```
Comment (wusui) [inline docs/intro_testers.rst]: Maybe we should add a line like: "Each subdirecory tree of https://github.com/ceph/ceph-qa-suite constitutes a `suite`"

File: docs/intro_testers.rst [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+Scheduling
+==========
+Most testing happens by scheduling `runs`. The way we do that is using the
+`teuthology-suite` command.
+
+To get a preview of what `teuthology-suite` might do, try::
+
+    teuthology-suite -v -s smoke -c master -m mira --dry-run
+
```
Comment (wusui) [inline docs/intro_testers.rst]: We should probably explain here that the -m mira option in this example specifies name of the set of machines from which teuthology-suite is selecting.

File: docs/README.rst [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+Each suite name is determined by the name of the directory in ``ceph-qa-suite``
+that contains that suite. The directory contains subdirectories and yaml files,
+which, when assembled, produce valid tests that can be run. The test suite
+application generates combinations of these files and thus ends up running a
+set of tests based off the data in the directory for the suite.
+
+To run a suite, enter::
+
+    teuthology-suite -s <suite> [-c <ceph>] [-k <kernel>] [-e email] [-f flavor] [-t <teuth>] [-m <mtype>]
+
```
Comment (wusui) [inline docs/README.rst]: Is [templates] missing here?

File: docs/README.rst [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+cuttlefish kernel, with a 'basic' kernel flavor, and the teuthology
+cuttlefish branch will be used.  It will run on plana machines and send an email
+to bob.smith@foo.com when it's completed. For more details on
+``teuthology-suite``, please consult the output of ``teuthology-suite --help``.
+
+In order for a queued task to be run, a teuthworker thread on
+``teuthology.front.sepia.ceph.com`` needs to remove the task from the queue.
+On ``teuthology.front.sepia.ceph.com``, run ``ps aux | grep teuthology-worker``
+to view currently running tasks. If no processes are reading from the test
+version that you are running, additonal teuthworker tasks need to be started.
```
Comment (wusui) [inline docs/README.rst]: The teuthology.front.sepia.ceph.com references may be confusing to downstream users.

File: docs/downburst_vms.rst [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+=============
+
+Teuthology also supports virtual machines via `downburst
+<https://github.com/ceph/downburst>`__, which can function like physical
+machines but differ in the following ways:
+
+VPS Hosts:
+--------
+The following description is based on the Red Hat lab used by the Ceph
+development and quality assurance teams.
```
Comment (wusui) [inline docs/downburst_vms.rst]: and upstream quality assurance teams

File: docs/README.rst (hunk 3 above)
Comment (zmc) [inline docs/README.rst]: What do you mean by that?

File: docs/README.rst [hunk 6]
```diff
@@ -0,0 +1,157 @@
...
+
+Detailed test configuration
+===========================
+
+See :ref:`detailed_test_config`.
+
+
+Virtual Machine Support
+=======================
+
```
Comment (vasukulkarni) [inline docs/README.rst]: I think it is worth mentioning here that teuthology can run on Openstack VM's and VMs on qemu/kvm which are called vps (vps is still confusing term)

File: docs/README.rst [hunk 7]
```diff
@@ -0,0 +1,157 @@
...
+Test Suites
+===========
+
+Most of the current teuthology test suite execution scripts automatically
+download their tests from the master branch of the appropriate github
+repository.  People who want to run experimental test suites usually modify the
+download method in the ``teuthology/task`` script to use some other branch or
+repository. This should be generalized in later teuthology releases.
+Teuthology QA suites can be found in ``src/ceph-qa-suite``. Make sure that this
+directory exists in your source tree before running the test suites.
```
Comment (vasukulkarni) [inline docs/README.rst]: It is not mandatory to clone the ceph-qa-suite, if not done, teuthology will get the master suite in user's src folder

File: docs/README.rst [hunk 8]
```diff
@@ -0,0 +1,157 @@
...
+* ``teuth``: version of teuthology to run
+* ``mtype``: machine type of the run
+* ``templates``: template file used for further modifying the suite (optional)
+
+For example, consider::
+
+     teuthology-suite -s rbd -c wip-fix -k cuttlefish -e bob.smith@foo.com -f basic -t cuttlefish -m plana
+
+The above command runs the rbd suite using the wip-fix branch of ceph, the
+cuttlefish kernel, with a 'basic' kernel flavor, and the teuthology
```
Comment (vasukulkarni) [inline docs/README.rst]: cuttlefish is too old, how about mentioning jewel now?

File: docs/detailed_test_config.rst [hunk 9]
```diff
@@ -0,0 +1,271 @@
...
+  behaving like one.
+  Sub-tasks can nest further information.  For example, overrides
+  of install tasks are project specific, so the following section of a yaml
+  file would cause all ceph installations to default to using the cuttlefish
+  branch::
+
+    overrides:
+      install:
+        ceph:
+          branch: cuttlefish
```
Comment (vasukulkarni) [inline docs/detailed_test_config.rst]: it could use 'jewel' as an example branch for new comers.

File: docs/detailed_test_config.rst [hunk 10]
```diff
@@ -0,0 +1,271 @@
...
+The above list is a very incomplete description of the tasks available on
+teuthology. The teuthology/task subdirectory contains the teuthology-specific 
+python files that implement tasks.
+
+Extra tasks used by teuthology can be found in ceph-qa-suite/tasks.  These
+tasks are not needed for teuthology to run, but do test specific independent
+features.  A user who wants to define a test for a new feature can implement
+new tasks in this directory.
+
+Many of these tasks are used to run shell scripts that are defined in the
```
Comment (vasukulkarni) [inline docs/detailed_test_config.rst]: s/shell/python.

File: docs/downburst_vms.rst [hunk 11]
```diff
@@ -0,0 +1,117 @@
...
+When a virtual machine is locked, downburst is run on that machine to install a
+new image.  This allows the user to set different virtual OSes to be installed
+on the newly created virtual machine.  Currently the default virtual machine is
+ubuntu (precise).  A different vm installation can be set using the
+``--os-type`` and ``--os-version`` options in ``teuthology.lock``.
+
+When a virtual machine is unlocked, downburst destroys the image on the
+machine.
+
+Temporary yaml files are used to downburst a virtual machine.  A typical
```
Comment (vasukulkarni) [inline docs/downburst_vms.rst]: I think this is not user configurable and internal yaml? If so we could delete this?

File: docs/README.rst (hunk 6 above)
Comment (zmc) [inline docs/README.rst]: I think that's what the next two lines do?

File: docs/README.rst (hunk 8 above)
Comment (zmc) [inline docs/README.rst]: Sure

File: docs/downburst_vms.rst (hunk 11 above)
Comment (zmc) [inline docs/downburst_vms.rst]: Sure, I think we can ditch it.

File: docs/README.rst (hunk 4 above)
Comment (zmc) [inline docs/README.rst]: True. A lot of references in these docs are specific to sepia; I'm not sure how to give useful examples without tying them to one lab, though.

File: docs/README.rst (hunk 7 above)
Comment (zmc) [inline docs/README.rst]: Yeah actually that whole paragraph is somewhat bogus. I'll just remove it.

File: docs/intro_testers.rst (hunk 2 above)
Comment (zmc) [inline docs/intro_testers.rst]: Done

File: docs/README.rst (hunk 3 above)
Comment (wusui) [inline docs/README.rst]: To be honest, I thought that this was a field that I saw in the docs but I can't find it now.  Please ignore for now.

**Comment by zmc** [comment]
I think this is ready to go, unless there are more changes people would like to see.

**Comment by vasukulkarni** [comment]
@zmc  LGTM

**Comment by wusui** [comment]
I'm cool with it.

**Pull request merged**